### PR TITLE
Refactoring Driver::S3 to reduce code complexity

### PR DESCRIPTION
**Pull request by awead**
@cam156 from today's refactoring session. We whittled down our ACB size and now have one less overly-long method.

### CODE REVIEW

File: lib/browse_everything/driver/s3.rb [hunk 1]
```diff
@@ -6,6 +6,8 @@ class S3 < Base
       DEFAULTS = { signed_url: true, region: 'us-east-1' }.freeze
       CONFIG_KEYS = [:app_key, :app_secret, :bucket].freeze
 
+      attr_reader :file_entries
```
Comment (cam156) [inline lib/browse_everything/driver/s3.rb]: @awead the name file_entries is indicative of just files being listed to me should it be file_and_path_entries, or directory_listing_entires?

File: lib/browse_everything/driver/s3.rb [hunk 2]
```diff
@@ -20,23 +22,35 @@ def validate_config
         raise BrowseEverything::InitializationError, "Amazon S3 driver requires #{CONFIG_KEYS.join(',')}"
       end
 
+      # @return [Array<BrowseEverything::FileEntry>]
```
Comment (cam156) [inline lib/browse_everything/driver/s3.rb]: Yeah!  Yard docs!!! 👏

File: lib/browse_everything/driver/s3.rb (hunk 1 above)
Comment (awead) [inline lib/browse_everything/driver/s3.rb]: or bucket_entries, but yes, something that means "everything that is in this path"

**Comment by awead** [comment]
@cam156 I named it `file_entries` because they were all instances of BE::FileEntry. However, a FileEntry can be a file or a directory. So you make a good point! I renamed it to just "entries" to keep it inline with the class.

**Pull request merged**